lexers/mediawiki.lua

```diff
@@ -1,42 +1,72 @@
 -- Copyright 2006-2025 Mitchell. See LICENSE.
 -- MediaWiki LPeg lexer.
 -- Contributed by Alexander Misel.
-
-local lexer = require('lexer')
-local token, word_match = lexer.token, lexer.word_match
+local lexer = lexer
 local P, S, B = lpeg.P, lpeg.S, lpeg.B
+local lex = lexer.new(...)
 
-local lex = lexer.new('mediawiki')
+-- HTML-like tags
+local dq_str = P('"') * ((lexer.any - S('>"\\')) + ('\\' * lexer.any))^0 * P('"')
+local unquoted_attr = (lexer.any - (S('"' .. "'" .. '<>=') + lexer.space))^1
+local tag_attr = lex:tag(lexer.ATTRIBUTE, lexer.alpha^1 * lexer.space^0 *
+	('=' * lexer.space^0 * (dq_str + unquoted_attr))^-1 * lexer.space^0)
+local tag_name = lexer.alnum^1
+local tag_inner_content = lexer.space^0 * tag_attr^0 * lexer.space^0 * P('/')^-1
 
--- Comments.
-lex:add_rule('comment', token(lexer.COMMENT, lexer.range('<!--', '-->')))
+-- The tag rule should consume from '<' to the final '>'
+lex:add_rule('tag', lex:tag(lexer.TAG, '<' * P('/')^-1 * tag_name * tag_inner_content * P('>')))
 
--- HTML-like tags
-local tag_start = token(lexer.TAG, '<' * P('/')^-1 * lexer.alnum^1 * lexer.space^0)
-local dq_str = '"' * ((lexer.any - S('>"\\')) + ('\\' * lexer.any))^0 * '"'
-local tag_attr = token(lexer.ATTRIBUTE, lexer.alpha^1 * lexer.space^0 *
-	('=' * lexer.space^0 * (dq_str + (lexer.any - lexer.space - '>')^0)^-1)^0 * lexer.space^0)
-local tag_end = token(lexer.TAG, P('/')^-1 * '>')
-lex:add_rule('tag', tag_start * tag_attr^0 * tag_end)
-
--- Link
-lex:add_rule('link', token(lexer.STRING, S('[]')))
-lex:add_rule('internal_link', B('[[') * token(lexer.LINK, (lexer.any - '|' - ']]')^1))
-
--- Templates and parser functions.
-lex:add_rule('template', token(lexer.OPERATOR, S('{}')))
+-- Internal Link: [[Target]] or [[Target|Display Text]]
+-- The content can contain almost anything except unbalanced square brackets.
+-- We'll highlight the whole thing as LINK.
+local internal_link_content = (lexer.any - P(']]'))^1 -- Matches everything until ']]'
+lex:add_rule('internal_link', lex:tag(lexer.LINK, P('[[') * internal_link_content * P(']]')))
```

Comment:
I seem to recall asking something like this before, but is there a reason we cannot use `lexer.range('[[', ']]')`?